robot-core/src/main/java/org/obolibrary/robot/ExtractOperation.java

```diff
@@ -214,6 +237,143 @@ public static OWLOntology extract(
     }
   }
 
+  /**
+   * Extracts a materialized sub-ontology from the given ontology that only contains the given terms
+   * and the relations between them. The input ontology is not changed.
+   *
+   * @param inputOntology the ontology to extract from
+   * @param terms a set of IRIs for terms to extract
+   * @param outputIRI the OntologyIRI of the new ontology
+   * @param options map of extract options
+   * @param sourceMap map of term IRI to source IRI, or null (only used with annotate-with-source)
+   * @return a new ontology (with a new manager)
+   * @throws OWLOntologyCreationException on any OWLAPI problem
+   */
+  public static OWLOntology extractSubset(
+      OWLOntology inputOntology,
+      Set<IRI> terms,
+      IRI outputIRI,
+      Map<String, String> options,
+      Map<IRI, IRI> sourceMap)
+      throws OWLOntologyCreationException {
+    OWLOntology materializedOnt = materialize(inputOntology, outputIRI);
+    OWLOntology filteredOnt = filter(materializedOnt, terms, outputIRI);
+    copyPropertyAnnotations(inputOntology, filteredOnt);
+    ReduceOperation.reduce(filteredOnt, new org.semanticweb.elk.owlapi.ElkReasonerFactory());
+
+    return filteredOnt;
+  }
+
+  /**
+   * Materializes the given ontology using the relation-graph.Creates a new ontology that contains
+   * the input ontology axioms and the materialized axioms. The input ontology is not changed.
+   *
+   * @param ontology ontology to materialize.
+   * @param outputIRI IRI of the output ontology.
+   * @return materialized ontology.
+   * @throws OWLOntologyCreationException
+   */
+  private static OWLOntology materialize(OWLOntology ontology, IRI outputIRI)
+      throws OWLOntologyCreationException {
+    Config config = new Config(null, false, true, true, true, true, false);
+    Set<OWLClassAxiom> materializedAxioms =
+        RelationGraphUtil.computeRelationGraph(ontology, new HashSet<IRI>(), config);
```

Comment:
@hkir-dev you can pass the desired relations to compute here, instead of the empty set.

Reply:
Fixed the code accordingly. Filtered the object properties from the seed terms and passed to the relation-graph.

Reply:
@balhoff can you resolve this comment if addressed to your satisfaction?

Reply:
Thx